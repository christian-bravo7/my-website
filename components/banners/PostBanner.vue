<template>
  <section
    class="flex | relative | bg-black"
  >
    <img
      class="absolute inset-0 | w-full h-full | object-cover object-center | opacity-50"
      :src="image"
      alt="ballon"
    >
    <div class="flex flex-col | z-10 | container max-w-7xl w-full | mx-auto px-4 py-52">
      <h1 class="text-3xl md:text-5xl | font-bold | text-gray-50 | mb-6">
        {{ title }}
      </h1>
      <div class="flex justify-between items-center">
        <div class="flex flex-col">
          <span class="text-gray-200 | mb-4">{{ formattedDate }}</span>
          <div>
            <AppPill text="Welcome" />
          </div>
        </div>
        <PostShareWithSocialNetworks :post-information="{ title, description, url }" />
      </div>
    </div>
  </section>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator';
import dayjs from 'dayjs';

@Component
export default class BlogBanner extends Vue {
  @Prop({ type: String, required: true })
  readonly image!: string;

  @Prop({ type: Array, required: true })
  readonly date!: Array<number>;

  @Prop({ type: String, required: true })
  readonly title!: string;

  @Prop({ type: String, required: true })
  readonly description!: string;

  @Prop({ type: String, required: true })
  readonly url!: string;

  @Prop({ type: Number, required: true })
  readonly minutes!: number;

  get formattedDate (): string {
    return dayjs(this.date as any).format('MMM DD, YYYY');
  }
}
</script>
