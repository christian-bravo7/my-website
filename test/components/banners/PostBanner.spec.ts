import { mount } from '@vue/test-utils';

import PostBanner from '@/components/banners/PostBanner.vue';

describe('PostBanner component', () => {
  it('should render correclty', () => {
    const wrapper = mount(PostBanner, {
      propsData: {
        image: 'image.png',
        date: [2020, 10, 24],
        title: 'my new post',
        description: 'welcome to my new post',
        minutes: 2,
        url: 'localhost/my-new-post',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should format created date', () => {
    const wrapper = mount(PostBanner, {
      propsData: {
        image: 'image.png',
        date: [2020, 10, 24],
        title: 'my new post',
        description: 'welcome to my new post',
        minutes: 2,
        url: 'localhost/my-new-post',
      },
    });

    const date = wrapper.find('span.text-gray-200');

    expect(date.text()).toBe('Oct 24, 2020');
  });
});
