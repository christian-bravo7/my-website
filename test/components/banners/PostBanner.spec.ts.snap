// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostBanner component should render correclty 1`] = `
<section class="flex | relative | bg-black"><img src="image.png" alt="ballon" class="absolute inset-0 | w-full h-full | object-cover object-center | opacity-50">
  <div class="flex flex-col | z-10 | container max-w-7xl w-full | mx-auto px-4 py-52">
    <h1 class="text-3xl md:text-5xl | font-bold | text-gray-50 | mb-6">
      my new post
    </h1>
    <div class="flex justify-between items-center">
      <div class="flex flex-col"><span class="text-gray-200 | mb-4">Oct 24, 2020</span>
        <div><span class="py-1 px-2 text-xs font-bold mr-2 select-none capitalize rounded-full bg-pink-500 text-gray-50">
  Welcome
</span></div>
      </div>
      <div class="flex"><a target="_blank" href="https://www.linkedin.com/sharing/share-offsite/?url=localhost/my-new-post" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon ml-4 | text-gray-50"><svg></svg></a><a target="_blank" href="https://www.twitter.com/intent/tweet?url=localhost/my-new-post&amp;text=my new post" class="w-8 h-8 md:w-12 md:h-12 | p-1 | bg-transparent dark:text-gray-50 dark-hover:bg-blue-500 dark-hover:border-blue-500 text-gray-900 hover:bg-pink-500 hover:text-gray-50 hover:border-pink-500 rounded-full social-icon ml-4 | text-gray-50"><svg></svg></a></div>
    </div>
  </div>
</section>
`;
